frontend/src/stores/filterState/FilterState.ts

```diff
@@ -36,14 +36,167 @@ export interface FilterLineState {
 /**
  * State of a filter
  */
-export interface FilterState {
+export class FilterState {
+  constructor(
+    private _edges: FilterLineState[],
+    private _nodes: FilterLineState[]
+  ) {}
+
+  get edges(): FilterLineState[] {
+    // eslint-disable-next-line no-underscore-dangle
+    return this._edges;
+  }
+
+  get nodes(): FilterLineState[] {
+    // eslint-disable-next-line no-underscore-dangle
+    return this._nodes;
+  }
+
+  /**
+   * Gets the isActive value of a {@link FilterLineState}.
+   * @param filterLineType - filter line which is searched for
+   * @param entity - specifies whether this entity is a node or an edge
+   */
+  public getFilterLineIsActive(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): boolean {
+    const lineState: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    return lineState !== undefined ? lineState.isActive : false;
+  }
+
   /**
-   * States of the filter lines of the nodes
+   * Toggles the {@link isActive} property of the specified filter line.
+   * @param filterLineType - filter line which {@link isActive} property is toggled
+   * @param entity - specifies whether this entity is a node or an edge
    */
-  edges: FilterLineState[];
+  public toggleFilterLineActive(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
+      line.isActive = !line.isActive;
+    }
+  }
 
   /**
-   * States of the filter lines of the edges
+   * Sets the {@link isActive} property of the specified filter line active.
+   * @param filterLineType - filter line which {@link isActive} property is set active
+   * @param entity - specifies whether this entity is a node or an edge
    */
-  nodes: FilterLineState[];
+  public setFilterLineActive(
+    filterLineType: string,
+    entity: 'node' | 'edge'
+  ): void {
+    const line: FilterLineState | undefined = this.getLineFromFilterState(
+      filterLineType,
+      entity
+    );
+
+    if (line) {
```

Comment:
Can this be simplified to `line?.isActive = true`? Not sure on this. If not, never mind.